e2e/testclients/clientutils/util.go

```diff
@@ -44,55 +44,70 @@ type NotificationItem struct {
 	Persistent bool      `json:"persistent"`
 }
 
-type Content struct {
+type Event struct {
 	Message string `json:"message"`
 }
 
+type Receipt struct {
+	TxHash string         `json:"txHash"`
+	Result map[string]any `json:"result"`
+	Errors []string       `json:"errors"`
+}
+
 type NotificationCollection struct {
 	Notifications   []NotificationItem `json:"notifications"`
 	CacheableCursor string             `json:"cacheableCursor"`
 }
 
-func (c *NakamaClient) ListNotifications(k int) ([]*Content, error) {
+func (c *NakamaClient) ListNotifications(k int) ([]*Receipt, []*Event, error) {
 	path := "v2/notification"
 	options := fmt.Sprintf("limit=%d&cursor=%s", k, c.notificationCursor)
 	url := fmt.Sprintf("%s/%s?%s", c.addr, path, options)
 	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, url, nil)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	req.Header.Set("Authorization", c.authHeader)
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	defer resp.Body.Close()
 
 	bodyData, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	data := NotificationCollection{
 		Notifications:   make([]NotificationItem, 0),
 		CacheableCursor: "",
 	}
 	err = json.Unmarshal(bodyData, &data)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	c.notificationCursor = data.CacheableCursor
-	acc := make([]*Content, 0)
+	receipts := make([]*Receipt, 0)
+	events := make([]*Event, 0)
```

Comment:
Put an estimated number in here it prevents multiple allocation calls. It is a Trivial optimization and it doesn't always matter but good habit to have.